src/backend/graph.js

```diff
@@ -37,6 +39,31 @@ export class Graph {
     this._inEdges = {};
   }
 
+  equals(that: Graph): boolean {
+    function nodesSubset(a: Graph, b: Graph): boolean {
```

Comment:
why is `deepEqual(this._nodes, that._nodes) && deepEqual(this._edges, that._edges)` not sufficient?

Reply:
Good call: it is. Deep-diffing the full `Graph` objects is not sufficient, because `_inEdges` and `_outEdges` have undefined order. But the `_nodes` and `_edges` objects should be fine.